scripts/name_match.py

```diff
@@ -0,0 +1,177 @@
+            matched = True
+        elif cleaned_name.split()[-1] == cleaned_person_family_name:
+            # Example: Tom Brinkman
+            matched = True
+        elif cleaned_name.split()[0].replace(",", "") == cleaned_person_family_name:
+            # Example Kwan, Karen
+            matched = True
+        elif len(cleaned_name.split()) == 3 and (cleaned_person_family_name in cleaned_name.split()[1]):
+            # Example: Matt Huffman, M.
+            matched = True
+        elif len(cleaned_name.split()) == 3 and (cleaned_name.split()[0] == cleaned_person_family_name):
+            # Example: ZEIGLER of Montville
+            matched = True
+        elif len(cleaned_name.split()) == 4 and (cleaned_name.split()[2] in cleaned_person_family_name):
+            # Example: Louis W. Blessing, III
+            matched = True
+        elif len(cleaned_name.split()) > 4 and (cleaned_name.split()[1] in cleaned_person_family_name):
+            # Example: S. CHANG (Introduced by request of another party) in Hawaii
+            matched = True
+        elif cleaned_name.replace("'", "") == cleaned_person_family_name.replace("'", ""):
+            # Example O'Donnell
+            matched = True
+        elif cleaned_name.replace(" ", "") == cleaned_person_family_name.replace(" ", ""):
+            # Example Crosswhite Hader == CrosswhiteHader
+            matched = True
+    elif cleaned_name == cleaned_person_name:
+        matched = True
+    elif cleaned_name in cleaned_person_name:
+        matched = True
+    elif (" of " in cleaned_name) and (cleaned_name.split()[0] == cleaned_person_name.split()[-1]):
+        # Example: carpenter of aroostook
+        matched = True
+    elif cleaned_name.split()[0] == cleaned_person_name.split()[-1]: # Need to double check tests
+        # Another example for: Carpenter of Aroostook,
+        matched = True
+    elif len(cleaned_name.split()) == 2 and (cleaned_name.split()[0] == cleaned_person_name.split()[0]) and (cleaned_name.split()[-1] != cleaned_person_name.split()[-1]):
+        # 'Mark BLIER' and 'Mark Lawrence'
+        matched = False
+    elif len(cleaned_name.split()) > 0 and (cleaned_name.split()[0] == cleaned_person_name.split()[-1]):
+        # Example: West (Tammy)
+        matched = True
+    elif len(cleaned_name.split()) == 2 and (cleaned_name.split()[1] in cleaned_person_name):
+        #Matt Huffman, M.
+        matched = True
+    elif len(cleaned_name.split()) > 2 and (cleaned_name.split()[0] in cleaned_person_name):
+        # 'SANBORN, L. of Cumberland' and 'Victoria E. Morales'
+        matched = True
+
+    if matched:
+        return True
+    else:
+        return False
+
+
+@click.command()
+@click.argument("archive_data_csv")
+def entrypoint(archive_data_csv):
+    archive_data = []
+    jurisdiction = ""
+    with open(archive_data_csv, newline='') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
```

Comment:
since you aren't renaming the columns why not just

archive_data = list(reader)